src/db.rs

```diff
@@ -440,14 +452,18 @@ mod tests {
             ..Default::default()
         });
 
-        let rows = store.iter_txid(b"abcdefgh".to_vec().into_boxed_slice());
+        let rows = store.iter_txid(*b"abcdefgh");
         assert_eq!(rows.collect::<Vec<_>>(), to_rows(&items[1..5]));
     }
 
-    fn to_rows(values: &[&[u8]]) -> Vec<Box<[u8]>> {
+    fn to_rows(values: &[&[u8]]) -> Vec<[u8; HASH_PREFIX_ROW_SIZE]> {
         values
             .iter()
-            .map(|v| v.to_vec().into_boxed_slice())
+            .map(|v| {
+                let mut row = [0; HASH_PREFIX_ROW_SIZE];
+                row[..v.len()].copy_from_slice(v);
+                row
```

Comment:
the test case there had keys of different sizes, this is to pad the hardcoded keys with zeros, maybe better to include the zeros in there instead of this copy_from_slice code?

Reply:
Ah, OK, that makes sense.